### DISCUSSION TITLE
Multiple Nested Tabs

### DISCUSSION THREAD

**Discussion by vikashatri** [serenity-is/Serenity#5607] [Help]
I need to create Nested Tabs inside a form. By Default  if I use Tab[] attribute then it converts all the Tab[] into single level tab panel. For ex: Let's say I have A, B and C three tabs and when someone clicks on B Tab inside it also I need to create some Tabs like P, Q and R. Please suggest how can this be achieved in Serenity?

**Comment by VictorTomaili** [comment]
Use panel with tabs inside your tabs. You can put anything there, including panel. Check this samples 
https://serenity.is/demo/BasicSamples/OtherFormInTab
https://serenity.is/demo/BasicSamples/EntityDialogAsPanel/11077

**Reply by vikashatri** [reply]
Thanks for replying @VictorTomaili . But I need to create a custom Attribute something like [NestedTab] in C# So that it can be reused in the whole application. Any suggestions on how to achieve this will be much appreciated.

**Reply by VictorTomaili** [reply]
Create own attribute for nested tabs. Create a custom [PropertyProcessor](https://github.com/serenity-is/Serenity/blob/master/src/Serenity.Net.Entity/PropertyGrid/BasicPropertyProcessor/BasicPropertyProcessor.Tab.cs) what check attribute of form properties.
Then create own base [PropertyDialog](https://github.com/serenity-is/Serenity/blob/master/src/Serenity.Scripts/CoreLib/UI/Dialogs/PropertyDialog.ts) what overrides [initPropertyGrid](https://github.com/serenity-is/Serenity/blob/master/src/Serenity.Scripts/CoreLib/UI/Dialogs/PropertyDialog.ts#L67). Create own base [PropertyGrid](https://github.com/serenity-is/Serenity/blob/master/src/Serenity.Scripts/CoreLib/UI/Widgets/PropertyGrid.ts) what implement your attribute data.

**Reply by vikashatri** [reply]
Can you please suggest how to create and register own PropertyGrid in the Serenity Framework and how to register own PropertyProcessor and PropertyDialog after overriding?

**Reply by VictorTomaili** [reply]
I already show way to do. If you need extra support, you can contact us via support@serenity.is for custom implementations etc.
